File: Harvard_CS50/Week_1/mario.c
#include <stdio.h>
#include <cs50.h>

void pyramid(int n) {
    int null_space, i, blocks;
    null_space = n;
    i = (n-n);
    blocks = (n-n);

    for (; i < n;)
        {
            i++;
            for (; null_space > (n-n); null_space -= 1)
                {
                    printf(" ");
                }
            null_space = n - i;
            blocks += null_space;
            for (; blocks < n; blocks += 1)
                {
                    printf("#");
                }
            printf("  ");
            blocks -= null_space;
            for (; blocks > (n-n); blocks --)
                {
                    printf("#");
                }
            printf(" \n");
            blocks = (n-n);
        }
}

int main(void){
    int n;
    do {
        n = get_int("height: ");
    } while (n <= 0 || n >= 9);
    pyramid(n);




}

File: Harvard_CS50/Week_6/problem_set_6/mario.py
#Program which prints 2 mario pyramids
def main():

    #initial grab of height
    height = int(input("Enter Pyramid Height: "))
    
    #Ensure valid height size
    while height > 8 or height < 1:
        print("Invalid Height. Try Again.")
        height = int(input("Enter Pyramid Height: "))
    
    #prints the pyramid building in ascending order and then pyramid in descending  
    pyramid(height)

#definition of function to print left (ascending) pyramid
def pyramid(height):
    blanks = height
    blocks = 1
        
    while blanks > 0:
        print((" " * (blanks - 1)) + ("#" * blocks) + "  " + ("#" * blocks)) #+ (" " * (blanks - 1))
        blanks-=1
        blocks+=1

main()
